hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -649,14 +649,18 @@ Typical programming sequences are explained at the end of the Programmer's guide
 The OTP controller initializes automatically upon power-up and is fully operational by the time the processor boots.
 The only initialization steps that SW should perform are:
 
-1. Check that the OTP controller has successfully initialized by making sure that {{< regref STATUS >}} is IDLE and not in an ERROR state.
-2. Program and lock down the frequency of periodic background checks in HW.
+1. Check that the OTP controller has successfully initialized by reading {{< regref STATUS >}}. I.e., make sure that none of the ERROR bits are set, and that the DAI is idle ({{< regref STATUS.DAI_IDLE >}}).
```

Comment:
@mcy FYI

Reply:
this introduces the following changes with respect to the previous version:
- the maximum period is now defined using a bitmask instead of an exponent, since it is easier to handle like that in HW.
- the checks are opt-in now (previously, they have been enabled by default)
- the checks can now be triggered on-demand as well for scenarios where the periodic background checks are left disabled.
- an optional timeout counter is added to ward against physical tampering attempts that leave the OTP-internal bus in a wedged state. 

these changes have been introduced since there may be a need to leave the automatic background checks disabled if it later turns out that they are 1) flaky for some reason (they shouldn't be), 2) they become a reliability issue since they may lead to more wear of the physical OTP device than anticipated (again this shouldn't be the case, but we want to stay on the safe side and provide this option). 

if mask ROM is going to set this up, it may make sense to put this configuration into `OWNER_SW_CFG`. 
WDYT?

Reply:
What's the rationale for opt-in vs opt-out?

Reply:
I think this is mainly because we've handled it like that in the past. If you think opt-out would be better, I can change that.

Reply:
I'd be interested in the comments of others (@mcy, @tjaychen, @sjgitty) but opt-out seems safer because we want the checks to run.  There may be practical considerations that I'm not thinking of though, but you addressed the 2 main things above and I think we should have sufficient confidence in the verification to deal with (1), and in our analysis of the OTP macro for (2).

Reply:
after some more analysis of the OTP macro, I can confirm that (2) should not be an issue. and as you say, we should be able to deal with (1). hence, these two points basically become obsolete, and we could move to opt-out...

Reply:
sorry chiming back in.  Is this related to the periodic checks? I think I initially recommended opt-in because i was a tad worried about unexpected in-field conditions that might cause things to fail.  Without a way to skip the checks, we may effectively end up bricking the devices (at least until the next reboot).

IMO, in the end, we will probably enable this check based on other information (hard-coded into image / based on some soft OTP).  I felt okay with this since the initial boot still goes through these checks, just the periodic-ness is an opt-in.

Reply:
OK let's leave this as opt-in then.